File: 0392-is-subsequence/0392-is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        
        if (s.length() == 0) return true;
        
        Stack<Character> stack = new Stack<>();
        
        for (char c: t.toCharArray()) stack.push(c);
        
        int ptr = s.length() - 1;
        
        while (!stack.isEmpty()) {
            if (stack.pop() == s.charAt(ptr)) ptr--;
            if (ptr < 0) return true;
        }
        
        return false;
    }
}

File: 0392-is-subsequence/0392-is-subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int ti = 0;
        for (int si = 0; si < s.size(); si++) {
            
            while (ti < t.size() && t[ti] != s[si]) {
          
                ti++;
            }
            if (ti == t.size() && si < s.size()) {

                return false;
            }
            ti++;
        }
        return true;
    }
};